dot/digest/block_import_test.go

```diff
@@ -0,0 +1,494 @@
+				epochStateMock := NewMockEpochState(ctrl)
+
+				for _, consensusDigest := range digestData {
+					expectedBabeConsensusDigest := types.NewBabeConsensusDigest()
+					err := scale.Unmarshal(consensusDigest.Data, &expectedBabeConsensusDigest)
+					require.NoError(t, err)
+
+					epochStateMock.EXPECT().
+						HandleBABEDigest(header, expectedBabeConsensusDigest).
+						Return(nil)
+				}
+
+				return epochStateMock
+			},
+			createBlockHeader: func(t *testing.T) (*types.Header, []types.ConsensusDigest) {
+				_, _, genesisHeader := newWestendDevGenesisWithTrieAndHeader(t)
+				keyring, _ := keystore.NewSr25519Keyring()
+
+				keyPairs := []*sr25519.Keypair{
+					keyring.KeyAlice, keyring.KeyBob, keyring.KeyCharlie,
+				}
+
+				grandpaAuths := make([]types.GrandpaAuthoritiesRaw, len(keyPairs))
+				authorities := make([]types.AuthorityRaw, len(keyPairs))
+				for i, keyPair := range keyPairs {
+					authorities[i] = types.AuthorityRaw{
+						Key: keyPair.Public().(*sr25519.PublicKey).AsBytes(),
+					}
+
+					grandpaAuths[i] = types.GrandpaAuthoritiesRaw{
+						Key: keyPair.Public().(*sr25519.PublicKey).AsBytes(),
+					}
+				}
+
+				createNextEpoch := func() types.ConsensusDigest {
+					return createBABEConsensusDigest(t, types.NextEpochData{
+						Authorities: authorities[:],
+						Randomness:  [32]byte{0, 1, 2, 3, 4, 5, 6, 7, 8},
+					})
+				}
+
+				createNextConfigData := func() types.ConsensusDigest {
+					versionedNextConfigData := types.NewVersionedNextConfigData()
+					versionedNextConfigData.Set(types.NextConfigDataV1{
+						C1:             9,
+						C2:             10,
+						SecondarySlots: 1,
+					})
+
+					return createBABEConsensusDigest(t, versionedNextConfigData)
+				}
+
+				createScheduledChange := func() types.ConsensusDigest {
+					return createGRANDPAConsensusDigest(t, types.GrandpaScheduledChange{
+						Auths: grandpaAuths[:1],
+						Delay: 2,
+					})
+				}
+
+				return createBlockWithDigests(t, &genesisHeader,
+					createNextEpoch, createNextConfigData, createScheduledChange)
+			},
+		},
+	}
+
+	for tname, tt := range cases {
+		tt := tt
+
+		t.Run(tname, func(t *testing.T) {
+			importedHeader, consensusDigests := tt.createBlockHeader(t)
+
+			ctrl := gomock.NewController(t)
+			// idexes 0 and 1 belongs to the BABE digests next epoch data and next config data respectively
+			// the indexes after that are for GRANDPA scheduled change and forced change
+			epochStateMock := tt.setupEpochState(t, ctrl, importedHeader, consensusDigests[:2])
+			grandpaStateMock := tt.setupGrandpaState(t, ctrl, importedHeader, consensusDigests[2:])
+
+			onBlockImportDigestHandler := NewBlockImportHandler(epochStateMock, grandpaStateMock)
+			err := onBlockImportDigestHandler.Handle(importedHeader)
+			require.ErrorIs(t, err, tt.wantErr)
+			if tt.errString != "" {
+				require.EqualError(t, err, tt.errString)
+			}
+		})
+	}
+}
+
+type withDigest func() types.ConsensusDigest
```

Comment:
nit/ is there a more clear name for this type? It is a bit unclear to me without context what it is